standalone/docker-entrypoint.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+set -e
+
+cd ..
+. jmvenv/bin/activate
+cd scripts
+
+export JM_onion_serving_host="$(/sbin/ip route|awk '/src/ { print $9 }')"
+
+# First we restore the default cfg as created by wallet-tool.py generate
+if ! [ -f "$CONFIG" ]; then
+    cp "$DEFAULT_CONFIG" "$CONFIG"
+fi
+
+if ! [ -f "$AUTO_START" ]; then
+    cp "$DEFAULT_AUTO_START" "$AUTO_START"
+fi
+
+# generate ssl certificates for jmwalletd
+if ! [ -f "${DATADIR}/ssl/key.pem" ]; then
+    subj="/C=US/ST=Utah/L=Lehi/O=Your Company, Inc./OU=IT/CN=example.com"
+    mkdir -p "${DATADIR}/ssl/" \
+      && pushd "$_" \
+      && openssl req -newkey rsa:4096 -x509 -sha256 -days 3650 -nodes -out cert.pem -keyout key.pem -subj "$subj" \
+      && popd
+fi
+
+# ensure 'logs' directory exists
+mkdir -p "${DATADIR}/logs"
+
+# auto start services
+while read p; do
+    [[ "$p" == "" ]] && continue
+    [[ "$p" == "#"* ]] && continue
+    echo "Auto start: $p"
+    file_path="/etc/supervisor/conf.d/$p.conf"
+    if [ -f "$file_path" ]; then
+      sed -i 's/autostart=false/autostart=true/g' $file_path
+    else
+      echo "$file_path not found"
+    fi
+done <$AUTO_START
+
+declare -A jmenv
+while IFS='=' read -r -d '' envkey parsedval; do
+    n="${envkey,,}" # lowercase
+    if [[ "$n" =  jm_* ]]; then
+        n="${n:3}" # drop jm_
+        jmenv[$n]=${!envkey} # reread environment variable - characters might have been dropped (e.g 'ending in =')
+    fi
+done < <(env -0)
+
+# ensure a wallet name is present
+jmenv['rpc_wallet_file']=${jmenv['rpc_wallet_file']:-'jm_webui_default'}
+
+# adapt 'blockchain_source' is missing and we're in regtest mode
+if [ "${jmenv['network']}" == "regtest" ] && [ "${jmenv['blockchain_source']}" == "" ]; then
+    jmenv['blockchain_source']='regtest'
+fi
+
+# there is no 'regtest' value for config 'network': make sure to use "testnet" in regtest mode
+if [ "${jmenv['network']}" == "regtest" ]; then
+    jmenv['network']='testnet'
+fi
+
+# For every env variable JM_FOO=BAR, replace the default configuration value of 'foo' by 'BAR'
+for key in ${!jmenv[@]}; do
+    val=${jmenv[${key}]}
+    sed -i "s/^$key =.*/$key = $val/g" "$CONFIG" || echo "Couldn't set : $key = $val, please modify $CONFIG manually"
+    sed -i "s/^#$key =.*/$key = $val/g" "$CONFIG" || echo "Couldn't set : $key = $val, please modify $CONFIG manually"
```

Comment:
This is connected to the issue above regarding `max_cj_fee_abs` and `max_cj_fee_rel`.
They are commented out in the generated default config `#max_cj_fee_abs = x`.

This is indeed something I have thought about and that's why this approach was not added to the fix supplied to btcpayserver's image (https://github.com/btcpayserver/dockerfile-deps/pull/39). However, we must either:
- Overwrite the commented out value (or append) - the current approach.
- Provide our own default config (with all pros and cons this brings)
- Do not set it and let the user make the initial setting (seems undesired)

If these values are not present, you cannot perform actions as a taker (https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/d98cb36fb422edefab969a911564987f7878f021/jmclient/jmclient/wallet_rpc.py#L842-L845) - otherwise you get an ConfigNotPresent error.
> some actions require configuration variables to proceed (related to fees, in particular); if those are not allowed to fall back to defaults

Reply:
> I understand that the config is newly generated, but it still feels brittle. What if the comment style of the generated config changes? I might be overthinking this, though sweat_smile

Better overthinking than missing important stuff. This feedback and another pair of eyes is really helpful.

I know this line in particular is not that great. Could not get `sed` to do want I wanted in one statement.
On the other hand, it clearyl highlights what is done to the config and commented out lines within it - so I though I might just leave it like it is.
Also: It will really only take lines like `#<var> =` into account. It is definitely not great, since it is not future proof - but works with the current default config.